torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3008,6 +3008,160 @@ struct to_ir {
+          }
+          return std::make_shared<SimpleValue>(
+              graph
+                  ->insertNode(graph->createDict(
+                      type->expect<DictType>()->getKeyType(),
+                      type->expect<DictType>()->getValueType(),
+                      {},
+                      {}))
+                  ->output());
+        }
+
+        // Special case logic for if we have a dict comprehension
+        if (!apply.inputs().empty() && apply.inputs()[0].kind() == TK_DICT_COMP) {
+          auto dc = DictComp(apply.inputs()[0]);
+          auto dc_node = emitDictComprehension(dc, type_hint);
+          NamedValue self = NamedValue(dc.range(), "self", dc_node);
+          for (auto kwarg : apply.attributes()) {
+            auto name = StringLiteral::create(kwarg.range(), kwarg.name().name());
+            auto k = emitExpr(name);
+            auto v = emitExpr(kwarg.value());
+            NamedValue input_k = NamedValue(kwarg.range(), "", k);
+            NamedValue input_v = NamedValue(kwarg.range(), "", v);
+          emitBuiltinCall(
+            kwarg.range(), *graph, aten::_set_item, {self, input_k, input_v}, {});
+          }
+          return std::make_shared<SimpleValue>(dc_node);
+        }
+
+        // We can't feasibly register all possible key x value
+        // combinations of new prim ops for the case that we use the
+        // constructor with a dict literal. It makes much more sense
+        // to transform the dict literal into a list of tuples so that
+        // we can use the existing constructors
+        if (!apply.inputs().empty() &&
+            apply.inputs()[0].kind() == TK_DICT_LITERAL) {
+          auto dict_lit = DictLiteral(apply.inputs()[0]);
+          std::vector<Expr> zipped;
+          zipped.reserve(dict_lit.key_inputs().size());
+          TORCH_INTERNAL_ASSERT(
+              dict_lit.key_inputs().size() == dict_lit.value_inputs().size());
+          for (auto key_it = dict_lit.key_inputs().begin(),
+                    val_it = dict_lit.value_inputs().begin();
+               key_it != dict_lit.key_inputs().end();
+               ++key_it, ++val_it) {
+            auto tuple_inputs =
+                List<Expr>::create(apply.range(), {*key_it, *val_it});
+            auto tuple = TupleLiteral::create(apply.range(), tuple_inputs);
+            zipped.push_back(tuple);
+          }
+          auto ll_values = List<Expr>::create(apply.range(), zipped);
+          auto ll = ListLiteral::create(apply.range(), ll_values);
+          auto expr_list = List<Expr>::create(apply.range(), {ll});
+          // Change `apply` to a new Apply node holding a list of
+          // tuples
+          apply = Apply::create(
+              apply.range(), apply.callee(), expr_list, apply.attributes());
+        }
+
+        // If we have kwargs to include, we'll take a similar approach
+        // to the above logic and standardize the Apply node
+        if (!apply.attributes().empty()) {
```

Comment:
from 3101 - 3164 looks great !! I would think the logic is general enough here so that we dont need some of the other special cases - just the `apply.inputs().empty() && apply.attributes().empty()` case. Could you try removing them and comment what doesnt work ?